pkg/base/linuxutils/fs/fs_helper.go

```diff
@@ -312,3 +315,24 @@ func (h *WrapFSImpl) Unmount(path string) error {
 
 	return err
 }
+
+// DriveHasData detect FS from the provided device using lsblk --output FSTYPE
+// Receives file path of the device as a string
+// Returns error if something went wrong
+func (h *WrapFSImpl) DriveHasData(device string) (bool, error) {
+	var (
+		cmd          = fmt.Sprintf(DetectFSCmdTmpl, device)
+		stdout       string
+		driveHasData bool
+		err          error
+	)
+	if stdout, _, err = h.e.RunCmd(cmd,
+		command.UseMetrics(true),
+		command.CmdName(strings.TrimSpace(fmt.Sprintf(DetectFSCmdTmpl, "")))); err != nil {
```

Comment:
You need to check not only FS, but:
1. Partition table
2. Partitions
3. FS on disk or FS signature
4. Disk is a PV for LVG

Reply:
2, 4 are detected in disocoverDataOnDrive function. 3 is implemented in this method. Isn't it enough to discover partitions? Why do we need to discover partition table?

Reply:
Implemented